zips/zip-0228.rst

```diff
@@ -2,12 +2,539 @@
+|                             |                          |                                                  |Specification [#protocol-txnencoding]_.                              |  
++-----------------------------+--------------------------+--------------------------------------------------+---------------------------------------------------------------------+
+|``sizeProofsOrchard``        |``proofsOrchard``         |``byte[sizeProofsOrchard]``                       |The aggregated zk-SNARK proof for all Actions in the Action Group.   |  
++-----------------------------+--------------------------+--------------------------------------------------+---------------------------------------------------------------------+
+|``4``                        |``timeLimit``             |``uint32``                                        |The block number (in the future) after which the Actions of the      |  
+|                             |                          |                                                  |Action Group become invalid and should be rejected by consensus.     |  
++-----------------------------+--------------------------+--------------------------------------------------+---------------------------------------------------------------------+
+
+.. figure:: ../rendered/assets/images/zip-0228-swap-bundle.png
+    :width: 1000px
+    :align: center
+    :figclass: align-center
+
+    An example Zcash transaction with a Swap Bundle (simplified, for reference only).
+
+Spend Authorization Signature Changes
+-------------------------------------
+
+The spend authorization signature is computed in the manner specified in Section 4.15 of the protocol specification [#protocol-spendauthsig]_, except that the signature is generated over the Action Group Hash, :math:`\mathsf{AGHash}`, which is defined to be
+:math:`\mathsf{AGHash} := \mathsf{BLAKE2b}\texttt{-}\mathsf{256}(\texttt{"ZcashActionGroup"}, \mathsf{preAuthActionGroup})`.
+
+This change ensures that no changes are required in the Action statement [#zip-0226-circuit-statement]_ to be proved.
+
+SIGHASH Computation Changes
+---------------------------
+
+The addition of Action Groups to the Zcash transaction as another level of abstraction requires a change in the way transactions are hashed to compute the SIGHASH in ZIP 244, specifically in the ``orchard_digest`` section [#zip-0244-orchard-digest]_.
+We update the structure of the ``orchard_digest`` component of the SIGHASH as follows::
+
+    orchard_digest
+    ├── orchard_action_groups_digest
+    │   ├── orchard_actions_compact_digest
+    │   ├── orchard_actions_memos_digest
+    │   ├── orchard_actions_noncompact_digest
+    │   ├── flagsOrchard
+    │   ├── anchorOrchard
+    │   └── timeLimit
+    └── valueBalanceOrchard
+
+T.4: orchard_digest
+```````````````````
+In the case that Orchard actions are present in the transaction, this digest is
```

Comment:
Hmm, I see the confusion. All OrchardZSA Actions, Swaps or otherwise, would go through Action Groups now, right? So maybe: 
`In the case that OrchardZSA Action Groups are present in the transaction` ?

Reply:
Made the above change.